src/contracts/MultiSigWallet.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+// MultiSigWallet v0.1
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract MultiSigWallet is Ownable {
+    using SafeERC20 for IERC20;
+
+    address[] public _signers;
+
+    mapping(address => bool) public _isSigner;
+    mapping(address => bool) public _signed;
+
+    function appendSigner(address[] memory addresses_) public onlyOwner {
+        for (uint256 i = 0; i < addresses_.length; i++) {
+            _signers.push(addresses_[i]);
+            _isSigner[addresses_[i]] = true;
+        }
+    }
+
+    function removeSigner(address removeAddr_) public onlyOwner {
+        bool isRemoved = false;
+        for (uint256 i = 0; i < _signers.length; i++) {
+            if (removeAddr_ == _signers[i]) {
+                _signers[i] = _signers[_signers.length - 1];
+                _isSigner[removeAddr_] = false;
+                isRemoved = true;
+                break;
+            }
+        }
+        if (isRemoved) {
+            _signers.pop();
+        }
+    }
+
+    function isSignApproved() public view returns (bool) {
```

Comment:
well.. I don't know exactly what purpose of this function is but don't we need to check the given address is signed or not in this function?

Reply:
This is multisig wallet, that functions are 
- Lock some tokens 
- To withdraw locked token, several members all need to sign.
- Only after that, owner can withdraw that tokens(if any one of member don't sign, then it cannot be withdrawn).
Above function is used for that.

Reply:
Okay, then. Thanks!

Reply:
I just wrote some general opinions.